C#/homework1/homework1/main.cs

```diff
@@ -0,0 +1,189 @@
+                try
+                {
+                    Menu();
+                    givOption = Convert.ToChar(Console.ReadLine());
+                    switch (givOption)
+                    {
+                        case 'Q': flag = false; break;
+                        case 'A':
+                            Dictionary<string, string> dictToCreate = new Dictionary<string, string> { };
+                            
+                            foreach (string x in event_properties)
+                            {
+                                string new_data;
+                                Console.WriteLine("Enter " + x +  " of the new Event:");
+
+                                if (x == "RestName")
+                                    Rests();
+
+                                if (x == "DateTime")
+                                {
+                                    string data = "";
+                                    foreach (string y in datesTimes)
+                                    {
+                                        Console.WriteLine($"Enter {y}:");
+                                        data += Console.ReadLine() + " ";
+                                    }
+                                    data += "00";
+                                    new_data = data ;
+                                }
+                                else
+                                    new_data = Console.ReadLine();
+                                dictToCreate.Add(x, new_data);
+                            }
+
+                            Event toAdd = new Event(dictToCreate);
+                            if (toAdd.Correct())
+                                if (allEvents.Present(toAdd.Id))
+                                {
+                                    Console.WriteLine("Event with such ID already exists");
+                                }
+                                else 
+                                {
+                                    allEvents.Add(toAdd);
+                                    Console.WriteLine("Event successfully added to collection");
+                                    allEvents.Overwrite();
+                                }
+                            else
+                            {
+                                Console.WriteLine("There were mistakes in your input:");
+                                toAdd.PrintErrors();
+                            }
+                            break;
+                        case 'D':
+                            Console.Write(allEvents.PrintAll());
+                            Console.WriteLine("Select the Event to delete:");
+                            string to_delete = Console.ReadLine();
+                            Console.WriteLine(allEvents.DeleteOne(to_delete));
+                            allEvents.Overwrite(); break;
+                        case 'E':
+                            Console.Write(allEvents.PrintAll());
```

Comment:
Case "E" and "A" almost copy-paste. Fix it.

Reply:
done